addons/website/static/src/snippets/s_instagram_page/instagram_page.js

```diff
@@ -0,0 +1,94 @@
+import { Interaction } from "@website/core/interaction";
+import { registry } from "@web/core/registry";
+
+import { _t } from "@web/core/l10n/translation";
+
+// Note that Instagram can automatically detect the language of the user and
+// translate the embed.
+
+export class InstagramPage extends Interaction {
+    static selector = ".s_instagram_page";
+    dynamicContent = {
+        // We have to setup the message listener before setting the src, because
+        // the iframe can send a message before this JS is fully loaded.
+        _window: {
+            "t-on-message": this.onMessage,
+        },
+    }
+
+    setup() {
+        this.renderIframe();
```

Comment:
As you wish, but `renderIframe` is not really needed, everything could be moved directly into the `setup`